### PR TITLE
Revert "Revert "When running in CI, scrub e2e tests output to avoid leaking keys""

### PR DESCRIPTION
**Pull request by KevinFairise2**
Reverts DataDog/datadog-agent#32027

Tried to make an e2e test fail: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/735205063

### CODE REVIEW

**Review by chouetz — APPROVED** [review]

File: tasks/tools/gotest-scrubbed.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+### This script is used to run go test and scrub the output, the command can be used as follow:
+### ./gotest-scrubbed.sh <packages comma separated> -- <go tests flags>
+set -euo pipefail
```
Comment (pducolin) [inline tasks/tools/gotest-scrubbed.sh:4]: **💬 suggestion**
```suggestion
# bash strict mode, fail if go test fails
set -euo pipefail
```

Comment (pducolin) [reply tasks/tools/gotest-scrubbed.sh:4]: **❓ question**
Does it apply the pipe if go test fails?

**Review by pducolin — APPROVED** [review]

File: tasks/tools/gotest-scrubbed.sh (hunk 1 above)
Comment (KevinFairise2) [reply tasks/tools/gotest-scrubbed.sh:4]: Yeah that's a valid point. But I tested and it does. 
According to bash doc it always apply all the pipes, it only change the returned error value.
> If set, the return value of a pipeline is the value of the last (rightmost) command to exit with a non-zero status, or zero if all commands in the pipeline exit successfully

**Review by amenasria — APPROVED** [review]

**Pull request merged**